block/block.go

```diff
@@ -112,6 +113,30 @@ func (m *Manager) applyBlock(ctx context.Context, block *types.Block, commit *ty
 	return nil
 }
 
+func (m *Manager) checkPrevCachedBlocks(ctx context.Context) error {
```

Comment:
Probably best to create a mutex on the this function as can be called from both DA code and gossiped blocks callback which may lead to a Race condition.

Reply:
true. mutex added